jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -589,8 +686,8 @@ ecma_builtin_helper_string_prototype_object_index_of (ecma_string_t *original_st
       {
         break;
       }
-      if (ecma_builtin_helper_string_find_index (original_str_p, search_str_p, true,
-                                                 (ecma_length_t) start_ends_with, &index_of))
+      if (ecma_builtin_helper_string_find_index (original_str_p, search_str_p,
```

Comment:
I think it would make more sense to use `ecma_builtin_helper_string_find_last_index` for `ECMA_STRING_ENDS_WITH`.